src/HexagonOffload.cpp

```diff
@@ -378,10 +277,83 @@ Stmt inject_hexagon_rpc(Stmt s, const Target &host_target) {
         }
     }
 
-    InjectHexagonRpc injector(target);
+    Module hexagon_module("hexagon", target);
+    InjectHexagonRpc injector(hexagon_module);
     s = injector.inject(s);
+
+    if (!hexagon_module.functions().empty()) {
+        *submodules++ = hexagon_module;
```

Comment:
insert_iterators are weird. The increment operator is actually a nop so it could just be removed. Passing by ref or value makes no difference, as assigning to the insert_iterator updates its internal state.

I didn't pass a vector ref as I wanted to make it clear adding to the end is the only thing that is allowed. I'm switching this over to passing the Module anyway and I think I'll just have the code call the appropriate append method.